app/controllers/application_controller.rb

```diff
@@ -172,9 +172,27 @@ def forward_to_app_config_domain
     redirect_to URL.url(request.fullpath)
   end
 
+  def current_user
+    return super unless edge_cached_action?
+
+    return nil unless session_current_user_id.present?
+      message = "😱😱😱😱😱 You cannot use current_user in this action\
```

Comment:
Can we move this message out into a constant? We don't need to recreate this string on every method invocation.